src/components/Share/index.js

```diff
@@ -0,0 +1,174 @@
+import { Modal, Button, AutoComplete, Select } from "antd";
+import {NotificationContainer, NotificationManager} from 'react-notifications';
+import { useSelector, useDispatch } from "react-redux";
+import { MdContactSupport } from "react-icons/md";
+import { IoSettingsOutline } from "react-icons/io5";
+import { SHOW_MODAL } from "../../action/types";
+import { handleShowModal } from "../../action";
+import "./styles.css";
+
+const emailOptions = [
+  {
+    value: "sandeep2016shah@gmail.com",
+  },
+  {
+    value: "sandeep@gamil.com",
+  },
+  {
+    value: "shah@gmail.com",
+  },
+  {
+    value: "kumar@gmail.com",
+  },
+];
+
+const Share = () => {
+  const dispatch = useDispatch();
+  const data = useSelector((state) => state);
```

Comment:
This will always lead to rerender as you are returning the state directly, consider your reducer, it always returns the new state and hence this useSelector will get new state each time anything has been changed, in this particular app rerendering might not be a big issue, but this is a code smell and going forward can create a problem.